servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -1876,14 +2008,22 @@ void executeDeletePolicy(ResourceContext ctx, String domainName, String policyNa
 
                 // extract the current policy for audit log purposes
 
-                Policy policy = getPolicy(con, domainName, policyName);
-                if (policy == null) {
+                List<Policy> policyVersions = new ArrayList<>();
```

Comment:
this should be moved to line 2017 so we're not wasting creating a new list if versions ends up being null or empty and we return 404

Reply:
Fixed